pages/login.js

```diff
@@ -12,12 +14,36 @@ class LoginPage extends Component {
       email: '',
       isError: false,
       message: null,
+      loading: false,
+      linkInvalid: false,
+      linkErrorMessage: null,
     };
 
     this.handleChange = this.handleChange.bind(this);
     this.loginCustomer = this.loginCustomer.bind(this);
   }
 
+  componentDidMount() {
+    // Get the 'token' from router
+    const { token } = this.props.router.query;
+
+    if(!token) {
+      return;
+    }
+
+    this.setState({ loading: true });
+
+    commerce.customer.getToken(token).then(() => {
+      Router.push('/account');
+    }).catch((error) => {
+      this.setState({
+        loading: false,
+        linkInvalid: true,
+        linkErrorMessage: error,
```

Comment:
```suggestion
        isError: true,
        message: ['The login link has expired. Please try again'],
```

This uses the existing error message functionality to give some feedback.

The message from the server is more technical. I think we can just assume that it's an expiry issue for this.

Reply:
Token expiry is the only legitimate error a user can expect. All other errors are likely to be caused by nefarious use of this API.

Reply:
ok so i shouldn't use `error.data.error.message` since it has the token id in it?

Reply:
Hard coding the message here is fine I think